tests/jballerina-unit-test/src/test/resources/test-src/expressions/checkedexpr/checked_error_return_type_mismatch_negative.bal

```diff
@@ -47,3 +47,14 @@ public function baz() {
 
     E1? res = wait w2;
 }
+
+type LhsErr distinct error;
+type LhsErrTwo distinct error;
+
+public function fooBar() returns LhsErr? {
+    int|LhsErrTwo x = check fbar();
+}
+
+function fbar() returns int|error {
+    return error("");
+}
```

Comment:
Should we also test something like

```ballerina
public function test() returns LhsErr? {
    int|LhsErr x = check func();
    int y = check func();
    int|LhsErrTwo z = check func();
}

function func() returns LhsErrTwo|int {
    return error("");
}
```

to verify the type ID check in the same type check?